### PR TITLE
Add pytest option to clean up test output

### PR DESCRIPTION
**Pull request by bojanserafimov**
Resolves https://github.com/neondatabase/neon/issues/1984

### CODE REVIEW

**Comment by hlinnaka** [comment]
The new option is disabled by default, and there are no changes to the github workflow, so this is just for convenience when running the tests locally?

I look at the logs in CI sometimes even for passing tests. Would they still be included in the allure report? (If we don't use this in the CI, I guess that question is moot..)

**Comment by SomeoneToIgnore** [comment]
> this is just for convenience when running the tests locally

When I've created the initial issue, it was caused by benchmark jobs generating too much data to fit into ~13GB limit on default GH Actions runners.

We don't use timeline and tenant data from Python tests usually (ever?), since we don't even store it in Allure, so presumably this could be enabled safely for CI runs.

File: test_runner/fixtures/neon_fixtures.py [hunk 1]
```diff
@@ -2661,7 +2673,10 @@ def test_output_dir(request: FixtureRequest, top_output_dir: Path) -> Iterator[P
 
     yield test_dir
 
-    allure_attach_from_dir(test_dir)
+    if pytestconfig.getoption("--cleanup-test-output"):
+        shutil.rmtree(test_dir)
```
Comment (SomeoneToIgnore) [inline test_runner/fixtures/neon_fixtures.py:2677]: Could we instead remove the layer and segment files, leaving logfiles and the rest intact?
This way, we will be able to remove most of the disk space and keep the data for Allure report, hence can turn the flag on by default.

**Comment by bojanserafimov** [comment]
> is just for convenience when running the tests locally?

Yes. It's annoying to run multiple tests, come back an hour later and see they've failed because I'm out of disk.

But anyway, I can make it CI-ready with the suggestions you brought up

**Pull request closed without merging**